io-engine/src/bdev/nexus/nexus_channel.rs

```diff
@@ -244,9 +283,25 @@ impl<'n> NexusChannel<'n> {
         &mut self,
         child_device: &str,
         reason: FaultReason,
-    ) {
+    ) -> Option<RebuildLogHandle> {
         self.nexus_mut()
-            .retire_child_device(child_device, reason, true);
+            .retire_child_device(child_device, reason, true)
+    }
+
+    /// Attaches a new rebuild log to write to.
+    pub(super) fn attach_rebuild_log(&mut self, log: RebuildLogHandle) {
+        if self
+            .rebuild_logs
+            .iter()
+            .all(|i| i.device_name() != log.device_name())
+        {
+            debug!(
+                "{self:?}: attaching rebuild log for '{dev}'",
+                dev = log.device_name()
```

Comment:
Hmm, I think you're right. It also bothers me that I have to clone the device name String each time I needed to access it, because of the lock (Mutex won't allow to return &str). I'll change keep a copy of dev name in the handle itself.

(But again, isn't it a proverbial premature optimization? The code path where it is accessed isn't really sensitive, the parking_lot mutex is a spinlock.)